engine/consensus/matching/engine.go

```diff
@@ -876,6 +835,71 @@ func (e *Engine) sealableResults() ([]*flow.IncorporatedResult, nextUnsealedResu
 	return results, nextUnsealeds, nil
 }
 
+// resultHasMultipleReceipts implements an additional TEMPORARY SAFETY measure:
+// only consider incorporatedResult sealable if we have AT LEAST 2 receipts committing
+// to the respective result from _different_ ENs
+func (e *Engine) resultHasMultipleReceipts(incorporatedResult *flow.IncorporatedResult) bool {
```

Comment:
The `resultHasMultipleReceipts` is inside of a for loop, we might be calling `e.receiptsDB.ByBlockIDAllExecutionReceipts` for the same block. Might be an optimization to group the results by block ID?

I think we should be good here, and can maybe optimize for later

Reply:
Good suggestion. I'll think about how we can add that optimization to update going to `master` (-> PR https://github.com/onflow/flow-go/pull/450)